### PR TITLE
Remove competition name from FMC filling sheet fix #352

### LINKED ISSUE #352: Remove competition name from reusable FMC sheets
When scrambles are provided separately, delegates can reuse FMC solution sheets across competitions (thus avoiding wasting unused ones). A simple solution would be to leave some space for competition name, as in the case of *round* and *attempt* number.

image

### PR DESCRIPTION
**Pull request by campos20**
This is how it should look

Edit

captura de tela de 2019-01-09 19-48-54

captura de tela de 2019-01-09 19-38-34

### CODE REVIEW

**Comment by jfly** [comment]
The code LGTM! I would like to get feedback from someone else before we merge this up, though. @lgarron what do you think about this idea?

**Comment by lgarron** [comment]
> @lgarron what do you think about this idea?

I strongly recommend changing it to "Competition: ____________" (make it two rows if needed). Otherwise, the purpose of the line is not discoverable, and just looks like a horizontal divider.

**Comment by jfly** [comment]
I'm down with that. @campos20, can you make that change? I know this will be a pain for the translations, but we can worry about that later.

**Comment by campos20** [comment]
> I'm down with that. @campos20, can you make that change? I know this will be a pain for the translations, but we can worry about that later.

Sure, I can! But how about completely removing it? Even the bar? Or do you think a delegate could confuse sheets between competitions?

**Comment by lgarron** [comment]
> Sure, I can! But how about completely removing it? Even the bar? Or do you think a delegate could confuse sheets between competitions?

I think it's pretty important to be able to identify the competition.

The competition name is also defined as one of the pieces of information to identify an attempt in the FMC overhaul draft PR: https://github.com/thewca/wca-regulations/pull/678/files

**Comment by campos20** [comment]
Done. Travis failing is expected. Also, the build is expected to fail until all translations are made.

**Comment by campos20** [comment]
Quite ugly, now that you mentioned. Align left for now?

captura de tela de 2018-12-26 00-29-33

**Comment by lgarron** [comment]
> Align left for now?

Yah, I think that would be good!

If you have more time, I'd also
- Add a few more mm of margin to the left of the text lines in that box.
- Give the "Competition" and "Competitor" fields lines two lines.

**Comment by campos20** [comment]
> * Add a few more mm of margin to the left of the text lines in that box.
> * Give the "Competition" and "Competitor" fields lines two lines.

I think this deserves a new PR, perhaps with the new rules for FMC. What you suggested is changing layout, I think we should split PRs for better control. We'll have to change the written rules on the sheet anyways (like 1 hour for writing your name and the solution), so perhaps you can merge this with align left and then you can change layout alongside with the new rules.

Btw, these changes are quite easy to do. Everything is organized on rects, which are easy to control.

**Comment by lgarron** [comment]
> I think this deserves a new PR, perhaps with the new rules for FMC.

Hmm, I don't follow. Could you explain how any FMC changes impact the FMC sheet?
(1 hour to write your name is not new, and only likely to get more lenient.)

Anyhow, a competition field change in one PR is fine, as long as it's aligned with the others. :-D

**Comment by campos20** [comment]
We'll have to change some lines of the written rules. I know this actually doesn't require changing the layout, but I was curious, do you have intention to change the layout after [The Overhaul](https://github.com/thewca/wca-regulations/pull/678)? If you have this intention, like changing the allowed moves, then we should perhaps make it all together. If you intend to keep the layout, then I think I agree with you.

**Comment by lgarron** [comment]
> I know this actually doesn't require changing the layout, but I was curious, do you have intention to change the layout after [The Overhaul](https://github.com/thewca/wca-regulations/pull/678)?

The overhaul PR is designed not to affect the FMC sheet! :-)

**Comment by campos20** [comment]
This is how it looks

captura de tela de 2019-01-07 16-10-40

The sheet with scramble continues the same.

**Comment by campos20** [comment]
https://github.com/thewca/tnoodle/pull/377/commits/802c2e42c9d61602600dd94baa481172af2d75a8 doesn't really change the layout, but it makes incredible easy to add more fields on competitor info. It also get rid of a lot of [obscure things](https://github.com/thewca/tnoodle/blob/cffbd90b05030f9ce6be7859a03abe12e984cb40/webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java#L637).

`offsetTop += fontSize + (int) (marginBottom*(withScramble ? 1 : 2.8));`

Everything looks the same (except that the spacing between competition name and competitor info is equal now)

captura de tela de 2019-01-09 18-28-18

captura de tela de 2019-01-09 18-28-28

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 1]
```diff
@@ -842,6 +842,21 @@ private static void fitAndShowText(PdfContentByte cb, String text, BaseFont bf,
         } while(true);
     }
     
+    private static void populateRect(PdfContentByte cb, Rectangle rect, ArrayList<String> list, ArrayList<Integer> alignList, BaseFont bf, int fontSize)  throws DocumentException {
+
```
Comment (jfly) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Can you add an assertion here that `list.size() == alignList.size()`?

Comment (campos20) [reply webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: Ok =)

**Comment by campos20** [comment]
> can you work on getting translations for the word "Competition" for all of our translations?

Sure!

**Pull request merged**